sympy/sets/tests/test_fancysets.py

```diff
@@ -441,6 +441,18 @@ def test_imageset_intersect_real():
     assert s.intersect(S.Reals) == imageset(Lambda(n, 2*n*pi - pi/4), S.Integers)
 
 
+def test_imageset_intersect_interval():
```

Comment:
Please add tests for the complex cases for example `Intersection(Lambda(n, n*pi + I), S.Integers))`.